src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs

```diff
@@ -2316,128 +2317,250 @@ public async Task<JArray> GetValuesFromDebuggerProxyAttribute(int objectId, int
+                    objects.TryAddRange(hiddenFieldsValues);
                 }
                 if (!getCommandType.HasFlag(GetObjectCommandOptions.WithProperties))
-                    return ret;
+                    return new JArray(objects.Values);
                 using var commandParamsObjWriter = new MonoBinaryWriter();
                 commandParamsObjWriter.WriteObj(new DotnetObjectId("object", objectId), this);
-                var props = await CreateJArrayForProperties(typeId[i], commandParamsObjWriter.GetParameterBuffer(), ret, getCommandType.HasFlag(GetObjectCommandOptions.ForDebuggerProxyAttribute), $"dotnet:object:{objectId}", i == 0, token);
-                ret = new JArray(ret.Union(props));
+                var props = await CreateJArrayForProperties(
+                    typeId,
+                    commandParamsObjWriter.GetParameterBuffer(),
+                    new JArray(objects.Values),
+                    getCommandType.HasFlag(GetObjectCommandOptions.ForDebuggerProxyAttribute),
+                    $"dotnet:object:{objectId}",
+                    i == 0,
+                    token);
+                var properties = await GetProperties(props, typeId, token);
+                objects.TryAddRange(properties);
 
                 // ownProperties
                 // Note: ownProperties should mean that we return members of the klass itself,
                 // but we are going to ignore that here, because otherwise vscode/chrome don't
                 // seem to ask for inherited fields at all.
                 //if (ownProperties)
-                    //break;
+                //break;
                 /*if (accessorPropertiesOnly)
                     break;*/
             }
             if (getCommandType.HasFlag(GetObjectCommandOptions.AccessorPropertiesOnly))
             {
-                var retAfterRemove = new JArray();
-                List<List<FieldTypeClass>> allFields = new List<List<FieldTypeClass>>();
-                for (int i = 0; i < typeId.Count; i++)
+                var ownId = typeIdsIncludingParents[0];
+                var ownFields = await GetTypeFields(ownId, token);
+
+                var parentFields = new List<FieldTypeClass>();
+                for (int i = 1; i < typeIdsIncludingParents.Count; i++)
+                    parentFields.AddRange(await GetTypeFields(typeIdsIncludingParents[i], token));
+
+                var ownDuplicatedFields = ownFields.Where(field => objects.Any(obj =>
+                        field.Name.Equals(obj.Key) &&
+                        (obj.Value["isOwn"] == null ||
+                        obj.Value["isOwn"].Value<bool>() ||
+                        !obj.Value["isOwn"].Value<bool>())));
+
+                var parentDuplicatedFields = parentFields.Where(field => objects.Any(obj =>
+                        field.Name.Equals(obj.Key) &&
+                        (obj.Value["isOwn"] == null ||
+                        !obj.Value["isOwn"].Value<bool>())));
```

Comment:
Is it checking if there is no value for isOwn, or it's true or it's false?
Is it not it the same of removing the 3 checks?